pkg/sign/certificate.go

```diff
@@ -29,6 +29,10 @@ import (
 	"time"
 )
 
+type Certificate interface {
```

Comment:
Can we call this `CertificateAuthority`?

Reply:
Thinking more about this. For the Fulcio use-case `CertificateAuthority` makes sense. But consider a scenario where an existing certificate is provided (i.e it's already signed by a CA before artifact signing begins). In this case the name `CertificateAuthority` is a bit misleading. How about changing this to `CertificateProvider` or something similar?